licence-checker/licence-checker.py

```diff
@@ -50,7 +51,7 @@ def search_line(self, licence_first_word):
         return self.comment_prefix + ' ' + licence_first_word
```

Comment:
Can we call `re.escape` in `search_line`, rather than at its call site? That way, it returns the same format as `expected_full_line`, which I think is probably a bit cleaner. Also, would you mind writing a short docstring for this and `expected_full_line` to note that they return a regex?

In fact, thinking about it, we probably want to include the `re.compile` here too. (I have a horrible feeling that I suggested something else before: if so, I'm really sorry!). Maybe `search_line` and `expected_full_line` should return a `re.Pattern`?

Oh, and then maybe the methods should be renamed to something like `search_line_pattern` and `full_line_pattern`? (I don't have strong feelings about the exact names, but making it clear from the name that they return a pattern would be nice).

Reply:
😅 No worries. I've renamed the two methods to better reflect their design intent and added the docstring.

Reply:
😅 No worries. I like the last suggestion, I also re-named the methods and added the docstring as suggested!